humility-core/src/hubris.rs

```diff
@@ -5235,25 +5306,37 @@ impl HubrisEnum {
         }
     }
 
-    //
-    // This is a little... loose.  It compares one enum to another and
-    // determines if they have the same variant names, offsets, tags, etc.
-    // This can still be fooled, but it would have to be by enums that
-    // appear pretty similar to one another.  If it becomes time to make this
-    // correct, it will probably be a good occasion to rethink how we store
-    // type information.
-    //
-    pub fn approximately_same(&self, rhs: &HubrisEnum) -> bool {
-        for (l, r) in self.variants.iter().zip(rhs.variants.iter()) {
-            if l.name != r.name || l.offset != r.offset || l.tag != r.tag {
-                return false;
+    fn differs(&self, hubris: &HubrisArchive, rhs: &Self) -> Result<bool> {
+        if self.size != rhs.size
+            || self.discriminant != rhs.discriminant
+            || self.variants.len() != rhs.variants.len()
```

Comment:
Do we need to check `self.tag` here?  I'm not exactly sure what it's doing!

Reply:
That's only a temporary, so... no.